screeps-profiler.js

```diff
@@ -14,20 +14,36 @@ function setupProfiler() {
     profile(duration, filter) {
       setupMemory('profile', duration || 100, filter);
     },
+    background(duration, filter) {
+      setupMemory('background', duration || false, filter);
+    },
+    clear() {
+      var restart = !!Memory.profiler && Memory.profiler.type == 'background'
+      var filter = (restart && Memory.profiler.filter) ? Memory.profiler.filter : null
+      var duration = (restart && Memory.profiler.disableTick) ? Memory.profiler.disableTick - Game.time : null
+      resetMemory()
+      if(restart) {
+        setupMemory('background', duration, filter);
+      }
+    },
     reset: resetMemory,
+    output(numresults) {
+      return Profiler.output(numresults);
+    }
   };
 
   overloadCPUCalc();
 }
 
 function setupMemory(profileType, duration, filter) {
   resetMemory();
+  var disableTick = Number.isInteger(duration) ? Game.time + duration : false
```

Comment:
Rather than a false duration, maybe we could update the `disableTick` value each tick, just add 1 to it each time to trick the profiler.  This way none of the other logic needs to change.